lints/lint_dnsname_underscore_in_trd.go

```diff
@@ -0,0 +1,55 @@
+package lints
+
+import (
+	"github.com/weppos/publicsuffix-go/publicsuffix"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"strings"
+)
+
+type DNSNameUnderscoreInTRD struct {
+	// Internal data here
+}
+
+func (l *DNSNameUnderscoreInTRD) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameUnderscoreInTRD) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c)
+}
+
+func underscoreInTRD(domain string) bool {
+	domainName, err := publicsuffix.Parse(domain)
+	if err != nil {
```

Comment:
This function should really return an error and not just reduce this to a boolean value. These are not equivalent situations.